ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/MainActivity.kt

```diff
@@ -132,36 +201,64 @@ fun WearApp(watchRepository: WatchRepository) {
                     )
                 }
 
-                composable(Screen.WatchList.route) {
-                    WatchListScreen(
-                        scalingLazyListState = scalingLazyListState,
-                        watchRepository = watchRepository,
-                        showVignette = vignetteVisible,
-                        onClickVignetteToggle = { showVignette ->
-                            vignetteVisible = showVignette
-                        },
-                        onClickWatch = { id ->
-                            swipeDismissableNavController.navigate(
-                                route = Screen.WatchDetail.route + "/" + id,
-                            )
+                composable(
+                    route = Screen.WatchList.route,
+                    arguments = listOf(
+                        // In this case, the argument isn't part of the route, it's just attached
+                        // as information for the destination.
+                        navArgument(SCROLL_TYPE_NAV_ARGUMENT) {
+                            type = NavType.EnumType(DestinationScrollType::class.java)
+                            defaultValue = DestinationScrollType.SCALING_LAZY_COLUMN_SCROLLING
                         }
                     )
+                ) {
+                    val passedScrollType = it.arguments?.getSerializable(SCROLL_TYPE_NAV_ARGUMENT)
+
+                    if (passedScrollType == DestinationScrollType.SCALING_LAZY_COLUMN_SCROLLING) {
```

Comment:
Nit: Do we need the `if` statements here inside the `composable`? It might make sense to do a `check(passedScrollType == DestinationScrollType.SCALING_LAZY_COLUMN_SCROLLING)` if there's no reason it should ever be a different one versus a strange blank screen.

Reply:
Ah, yes, good catch, fixed.